overlord/ifacestate/ifacemgr.go

```diff
@@ -195,7 +208,21 @@ Run "systemctl enable --now snapd.apparmor" to correct this.`)
 }
 
 // Ensure implements StateManager.Ensure.
-func (m *InterfaceManager) Ensure() error {
+func (m *InterfaceManager) Ensure() (err error) {
+	if udevMonErr := m.ensureUDevMon(); udevMonErr != nil {
+		defer func() {
+			if err != nil {
+				err = fmt.Errorf("%w; %v", udevMonErr, err)
+			} else {
+				err = udevMonErr
+			}
+		}()
+	}
+	err = m.ensurePrompting()
```

Comment:
as discussed this can be simplified and go into StartUp (and the bit in Stop stays)

Reply:
the code will need some tests at this package level